provider/awssd/aws_sd.go

```diff
@@ -264,6 +316,220 @@ func (p *AWSSDProvider) updatesToCreates(changes *plan.Changes) (creates []*endp
 	return creates, deletes
 }
 
+func (p *AWSSDProvider) handleSRVChanges(namespaces []*sd.NamespaceSummary, creates []*endpoint.Endpoint, deletes []*endpoint.Endpoint) ([]*endpoint.Endpoint, []*endpoint.Endpoint, error) {
+	//create a struct of all relevant changes to handle SRV-related changes
+	type relevantChanges struct {
+		ACre   []*endpoint.Endpoint // max len is 1
+		SRVCre []*endpoint.Endpoint // max len is 1
+		SRVDel []*endpoint.Endpoint // max len is n
+	}
+
+	createsByNamespaceID := p.changesByNamespaceID(namespaces, creates)
+	deletesByNamespaceID := p.changesByNamespaceID(namespaces, deletes)
+
+	for _, ns := range namespaces {
+		nsID := *ns.Id
+		nsCreates, okNsCre := createsByNamespaceID[nsID]
+		nsDeletes, okNsDel := deletesByNamespaceID[nsID]
+		//loop all namespaces where we have at least 1 change
+		if !(okNsCre || okNsDel) {
+			continue
+		}
+
+		services, err := p.ListServicesByNamespaceID(aws.String(nsID))
+		if err != nil {
+			return nil, nil, err
+		}
+
+		//create a map of relevant changes by host. will loop this map later to determine any SRV-related changes
+		changesByHost := make(map[string]*relevantChanges)
+		if okNsCre {
```

Comment:
This condition is redundant. It should be safe to iterate an empty slice.

Reply:
fixed